src/zcbor_decode.c

```diff
@@ -1378,19 +1404,23 @@ bool zcbor_any_skip(zcbor_state_t *state, void *result)
 			/* fallthrough */
 		case ZCBOR_MAJOR_TYPE_LIST:
 			if (indefinite_length_array) {
-				state_copy.payload++;
 				value = ZCBOR_LARGE_ELEM_COUNT;
 			}
 			state_copy.elem_count = (size_t)value;
 			state_copy.decode_state.indefinite_length_array = indefinite_length_array;
 			while (!zcbor_array_at_end(&state_copy)) {
-				if (!zcbor_any_skip(&state_copy, NULL)) {
+				if (!traverse_element(&state_copy, do_print, indent_len + 1)) {
 					ZCBOR_FAIL();
 				}
 			}
-			if (indefinite_length_array && !array_end_expect(&state_copy)) {
-				ZCBOR_FAIL();
+			if (indefinite_length_array) {
+				state_copy.payload++; /* payload bounds checked by zcbor_array_at_end(). */
+			}
+#ifdef ZCBOR_DIAG_PRINT
+			if(do_print && indefinite_length_array) {
+				zcbor_print_end(major_type, indent_len);
 			}
+#endif
```

Comment:
How much effort would it be to wrap all the printing code inside this function to another function so that the only change to the traverse function was  "if printing call this function" instead of having to inline all of this?  We should in general not make code more complex (that is, less readable) for debugging purposes.

Reply:
Tried to improve this, take a look